hazelcast-client/src/main/java/com/hazelcast/client/HazelcastClientManager.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client;
+
+import java.util.HashSet;
+import java.util.Collection;
+import java.util.Collections;
+
+import com.hazelcast.util.EmptyStatement;
+
+import java.util.concurrent.ConcurrentMap;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.OutOfMemoryHandler;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.impl.HazelcastClientProxy;
+import com.hazelcast.instance.OutOfMemoryErrorDispatcher;
+import com.hazelcast.core.DuplicateInstanceNameException;
+import com.hazelcast.client.config.XmlClientConfigBuilder;
+import com.hazelcast.client.impl.HazelcastClientInstanceImpl;
+import com.hazelcast.client.impl.ClientConnectionManagerFactory;
+import com.hazelcast.client.impl.DefaultClientConnectionManagerFactory;
+
+public class HazelcastClientManager {
+    /***
+     * Instance for clientManagers
+     */
+    public static final HazelcastClientManager HAZELCAST_CLIENT_MANAGER = new HazelcastClientManager();
+
+    static {
+        OutOfMemoryErrorDispatcher.setClientHandler(new ClientOutOfMemoryHandler());
+    }
+
+    private static final ConcurrentMap<String, HazelcastClientProxy> CLIENTS
```

Comment:
Why is this field static since HazelcastClientManager already is a singleton. Using the singleton I can access the clients map anway;

```
HazelcastClientManager.INSTANCE.clients
```